operator/controllers/utils.go

```diff
@@ -215,6 +215,10 @@ func AgentEnv(instance *agentv1alpha1.Agent, agentGroup string) []corev1.EnvVar
 			Name:  "APERTURE_AGENT_SERVICE_DISCOVERY_KUBERNETES_ENABLED",
 			Value: "false",
 		})
+		envs = append(envs, corev1.EnvVar{
```

Comment:
We should update this value in the actual CR instance as well so that we don't end up having 2 different values, one set by defaulter webhook and one passed using environment variable.

Reply:
We update some CR instance fields in reconciliation loop already and make update call at the end of the loop